### ISSUE TITLE
Question regarding negative prompts.

### ISSUE THREAD

**Issue by ovalerio** [computational-cell-analytics/micro-sam#1190]
Hello Team,

I used both bounding boxes and negative prompts when creating the labels during the manual segmentation.

I have a question, where are the negative prompts stored after committing the labels? Are the negative prompts still available? 

I would like to keep the negative prompts to fine tune a model using my manually annotated frames.

Thanks!

**Comment by anwai98** [comment]
Hi @ovalerio,

Thanks for your interests in `micro-sam`.

In short: yes, negative prompts are stored when you commit labels (in the output HDF5 file under "prompts/{object_id}/point_labels", with 0 for negative and 1 for positive), BUT our standard finetuning pipeline does not use them.

My suggestion here would be to trust your annotation quality and go with the iterative prompting strategy for the finetuning. The model does a good job at deriving valid (positive and) negative point prompts to make sure the segmentation training works well.

Do you have a specific reason why you'd like to use your own negative point prompts? (if yes, please feel free to elaborate here)

**Comment by ovalerio** [comment]
Hi @anwai98 ,

Thanks for answering my question. 

I will increase the number of labels and follow the iterative prompting strategy that you recommend me above.

My use case is for segmenting worm pharynx structures from a microscopy dataset.  In a previous project I was using SAM2 (without fine-tuning) and I was using bounding boxes combined with the negative prompts to do the segmentation.

This is how the data looks like:

<img width="677" height="399" alt="Image" src="https://github.com/user-attachments/assets/644bdbf5-3c17-4633-b37f-9f350c1c597b" />

I'm only interested in the structure inside the bounding box, but the other similar looking processes are sometimes getting labeled too. That's where the idea of using the negative prompts was coming from.

<img width="682" height="385" alt="Image" src="https://github.com/user-attachments/assets/860c8ea3-7cae-44f4-a67c-60c53ce0e79b" />

Are there plans to add `SAM2` support to  `micro-sam`?

Thanks again!

**Comment by anwai98** [comment]
Hi @ovalerio,

Thanks for the detailed response!

> I will increase the number of labels and follow the iterative prompting strategy that you recommend me above.

This would definitely be the right way-to-go. Lemme know if you want anymore suggestions!

> I'm only interested in the structure inside the bounding box, but the other similar looking processes are sometimes getting labeled too. That's where the idea of using the negative prompts was coming from.

Gotcha. If your annotations look as expected, the model would learn to segment only the annotated structure, for you to perform precise automatic and interactive segmentation using your finetuned model.

> Are there plans to add `SAM2` support to `micro-sam`?

A short answer is definitely. There's something concrete cooking right now hehe. We need a couple more months to wrap up our integration (keep an eye on the upcoming releases)

Until then, let's make `micro-sam` work for you. And in case you have any questions, please let us know! 😁

**Comment by ovalerio** [comment]
Hi @anwai98,

I tried fine-tuning my pharynx model using 80 labels this time.  Originally I wanted to train for 100 epochs, but the fine tuning stopped because the model was not improving after 10 epochs. I was using the vit-t-lm model and mixed precision training, and I wanted to ask you if vit-t is a good choice for my data or you would advise the vit-b instead.

Another question: When I generated the masks, I noticed some square looking artifacts.  Could this be due to the tiling I was using (384,384)?   Which values will you recommend for the tile and halo?  All my images are 800 x 800 pixels:

```bash
identify 2026-02-05-14-26-50-851099-basler_7021.tiff
2026-02-05-14-26-50-851099-basler_7021.tiff TIFF 800x800 800x800+0+0 8-bit Grayscale Gray 640256B 0.000u 0:00.000
```

<img width="1037" height="836" alt="Image" src="https://github.com/user-attachments/assets/fef32aab-82ec-476c-93e1-0d80a8b2ce01" />

<img width="1039" height="835" alt="Image" src="https://github.com/user-attachments/assets/5a9609aa-c671-4aaa-91b8-407a5931473f" />

**Comment by anwai98** [comment]
Hi @ovalerio,

Can you tell us what's your raw input's intensity range? And if you are applying any normalization?

**Comment by ovalerio** [comment]
Hi @anwai98,

I'm not applying normalization. The histogram card shows the intensity values range between 0 and 20. 

Regarding those artifacts, I asked you about, I had good news to share. I changed the tiling to 512 x 512 and the halos to 128 x 128 and the artifacts went away.  

I'm very happy with the current results, therefore I will close the question and thanks again you and the team for coaching me throughout the journey.

<img width="1080" height="607" alt="Image" src="https://github.com/user-attachments/assets/b22a2ca3-00ff-417e-b274-43e35f363a9a" />

<img width="782" height="623" alt="Image" src="https://github.com/user-attachments/assets/425e4a1e-ca76-4891-be1a-c4de37952e42" />

**Comment by anwai98** [comment]
Hi @ovalerio,

That's great to hear. Let us know if you have any new questions in future. Happy to discuss!

And thanks for your interest in `micro-sam` once again!
